examples/run/sycl/full_chain_algorithm.sycl

```diff
@@ -85,69 +108,78 @@ full_chain_algorithm::full_chain_algorithm(
       m_host_mr(parent.m_host_mr),
-      m_device_mr(std::make_unique<vecmem::sycl::device_memory_resource>(
-          &(m_data->m_queue))),
-      m_cached_device_mr(
-          std::make_unique<vecmem::binary_page_memory_resource>(*m_device_mr)),
-      m_copy(&(m_data->m_queue)),
       m_det_descr(parent.m_det_descr),
-      m_device_det_descr(
+      m_device_det_descr{std::make_unique<silicon_detector_description::buffer>(
           static_cast<silicon_detector_description::buffer::size_type>(
               m_det_descr.get().size()),
-          *m_device_mr),
-      m_clusterization(memory_resource{*m_cached_device_mr, &m_host_mr}, m_copy,
-                       m_data->m_queue_wrapper, parent.m_clustering_config),
-      m_spacepoint_formation(memory_resource{*m_cached_device_mr, &m_host_mr},
-                             m_copy, m_data->m_queue_wrapper),
+          *(m_data->m_device_mr))},
+      m_detector(parent.m_detector),
+      m_device_detector{std::make_unique<host_detector_type::buffer_type>()},
+      m_clusterization(std::make_unique<clusterization_algorithm>(
+          memory_resource{*(m_data->m_cached_device_mr), &m_host_mr},
+          m_data->m_copy, m_data->m_queue_wrapper, parent.m_clustering_config)),
+      m_spacepoint_formation(
+          memory_resource{*(m_data->m_cached_device_mr), &m_host_mr},
+          m_data->m_copy, m_data->m_queue_wrapper),
       m_seeding(parent.m_finder_config, parent.m_grid_config,
                 parent.m_filter_config,
-                memory_resource{*m_cached_device_mr, &m_host_mr}, m_copy,
-                &(m_data->m_queue)),
+                memory_resource{*(m_data->m_cached_device_mr), &m_host_mr},
+                m_data->m_copy, &(m_data->m_queue)),
       m_track_parameter_estimation(
-          memory_resource{*m_cached_device_mr, &m_host_mr}, m_copy,
-          &(m_data->m_queue)),
+          memory_resource{*(m_data->m_cached_device_mr), &m_host_mr},
+          m_data->m_copy, &(m_data->m_queue)),
       m_clustering_config(parent.m_clustering_config),
       m_finder_config(parent.m_finder_config),
       m_grid_config(parent.m_grid_config),
       m_filter_config(parent.m_filter_config) {
 
     // Copy the detector (description) to the device.
-    m_copy(vecmem::get_data(m_det_descr.get()), m_device_det_descr)->wait();
+    m_data->m_copy(vecmem::get_data(m_det_descr.get()), *m_device_det_descr)
+        ->wait();
+    if (m_detector != nullptr) {
+        *m_device_detector =
+            detray::get_buffer(detray::get_data(*m_detector),
+                               *(m_data->m_device_mr), m_data->m_copy);
+        m_device_detector_view = detray::get_data(*m_device_detector);
+    }
 }
 
 full_chain_algorithm::~full_chain_algorithm() {
     // Need to ensure that objects would be deleted in the correct order.
-    delete m_data;
+    m_clusterization.reset();
+    m_device_det_descr.reset();
+    m_device_detector.reset();
+    m_data.reset();
```

Comment:
This destructor should be removed.

Reply:
The destructor was made `default`. It can't be removed, as it needs to delete `traccc::sycl::details::full_chain_algorithm_data`. Which can only be done in that source file.

But yeah, the custom destructor could be removed.